test/no-empty-string-alt.test.js

```diff
@@ -0,0 +1,84 @@
+const noEmptyStringAltRule = require("../src/rules/no-empty-string-alt");
+const runTest = require("./utils/run-test").runTest;
+
+describe("GH003: No Empty String Alt", () => {
+  describe("successes", () => {
+    test("inline", async () => {
+      const strings = [
+        "![Chart with a single root node reading 'Example'](https://user-images.githubusercontent.com/abcdef.png)",
+        "`![''](image.png)`", // code block
+        '`![""](image.png)`', // code block
+      ];
+
+      const results = await runTest(strings, noEmptyStringAltRule);
+      expect(results).toHaveLength(0);
+    });
+    test("html image", async () => {
+      const strings = [
+        '<img alt="A helpful description" src="https://user-images.githubusercontent.com/abcdef.png">',
+        "`<img alt='' src='image.png'`", // code block
+      ];
+
+      const results = await runTest(strings, noEmptyStringAltRule);
+      expect(results).toHaveLength(0);
+    });
+  });
+  describe("failures", () => {
```

Comment:
Not super important, but it might also be useful to have a test for two failures in one line, like:

```md
<img src="cat.png" alt="" /> <img src="dog.png" alt="" />
```

I think we do handle this correctly with the `g` flag and iterating over matches per line, but it probably would still be good to test it.

Reply:
Good idea! Done in https://github.com/github/markdownlint-github/pull/85/commits/8f80d4d8368154ad4c9a0e1b244aff285f9c84c0.